packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx

```diff
@@ -103,50 +110,228 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
   }
 
   return (
-    <Modal
-      open={true}
-      submit={handleSubmit}
-      close={onCancel}
-      size="max"
-      header={modalTitle}
-      cta="Save"
-      ctaEnabled={saveEnabled}
-      autoFocusSelector="#id-modal-identify-joins-heading"
-    >
-      <div className="my-2 ml-3 mr-3">
-        <div className="row">
-          <div className="col-12">
-            <Field label="Display Name" required {...form.register("name")} />
-            <Field
-              label="Description (optional)"
-              textarea
-              minRows={1}
-              {...form.register("description")}
-            />
-            <Field
-              label="Identifier Type"
-              options={identityTypes.map((i) => i.userIdType)}
-              required
-              {...form.register("userIdType")}
-            />
-            <StringArrayField
-              label="Dimension Columns"
-              value={userEnteredDimensions}
-              onChange={(dimensions) => {
-                form.setValue("dimensions", dimensions);
-              }}
-            />
-            <SQLInputField
-              userEnteredQuery={userEnteredQuery}
-              datasourceId={dataSource.id}
-              form={form}
-              requiredColumns={requiredColumns}
-              identityTypes={identityTypes}
-              queryType="experiment-assignment"
-            />
+    <>
+      {sqlOpen && dataSource && (
+        <EditSqlModal
+          close={() => setSqlOpen(false)}
+          datasourceId={dataSource.id || ""}
+          requiredColumns={requiredColumns}
+          value={userEnteredQuery}
+          save={async (userEnteredQuery) => {
+            form.setValue("query", userEnteredQuery);
+          }}
+          validateResponse={(result) => {
```

Comment:
I think it's probably worth just defining this as a function outside of the component body ... just for readability

Reply:
10-4! Just updated that in this [commit](https://github.com/growthbook/growthbook/pull/1370/commits/62ff5f0f7dcc3d5d2c86cb141b1e8e10825e1fa8) - thanks for your review!